core/templates/pages/topic-editor-page/services/topic-editor-state.service.ts

```diff
@@ -18,541 +18,556 @@
-        // subtopic, then the 'subtopicId' part of the id of all subsequent
-        // subtopic pages should be decremented to make it in sync with the
-        // their corresponding subtopic ids.
-        if (newIndex !== -1) {
-          _newSubtopicPageIds.splice(newIndex, 1);
-          for (var i = 0; i < _cachedSubtopicPages.length; i++) {
-            var newSubtopicId = _getSubtopicIdFromSubtopicPageId(
-              _cachedSubtopicPages[i].getId());
-            if (newSubtopicId > subtopicId) {
-              newSubtopicId--;
-              _cachedSubtopicPages[i].setId(
-                _getSubtopicPageId(topicId, newSubtopicId));
-            }
-          }
-          for (var i = 0; i < _newSubtopicPageIds.length; i++) {
-            var newSubtopicId = _getSubtopicIdFromSubtopicPageId(
-              _newSubtopicPageIds[i]);
-            if (newSubtopicId > subtopicId) {
-              newSubtopicId--;
-              _newSubtopicPageIds[i] = _getSubtopicPageId(
-                topicId, newSubtopicId);
-            }
-          }
+      }
+      for (let i = 0; i < this._newSubtopicPageIds.length; i++) {
+        let newSubtopicId = this._getSubtopicIdFromSubtopicPageId(
+          this._newSubtopicPageIds[i]);
+        if (newSubtopicId > subtopicId) {
+          newSubtopicId--;
+          this._newSubtopicPageIds[i] = this._getSubtopicPageId(
+            topicId, newSubtopicId);
         }
-      },
+      }
+    }
+  }
 
-      /**
+  /**
        * Sets the topic rights stored within this service, propogating
        * changes to all bindings to the topic returned by
        * getTopicRights().
        */
-      setTopicRights: function(topicRights) {
-        _setTopicRights(topicRights);
-      },
+  setTopicRights(topicRights: TopicRights): void {
+    this._setTopicRights(topicRights);
+  }
 
 
-      /**
-       * Attempts to save the current topic given a commit message. This
-       * function cannot be called until after a topic has been initialized
-       * in this service. Returns false if a save is not performed due to no
-       * changes pending, or true if otherwise. This function, upon success,
-       * will clear the UndoRedoService of pending changes. This function also
-       * shares behavior with setTopic(), when it succeeds.
-       */
-      saveTopic: function(commitMessage, successCallback) {
-        if (!_topicIsInitialized) {
-          AlertsService.fatalWarning(
-            'Cannot save a topic before one is loaded.');
-        }
+  /**
+   * Attempts to save the current topic given a commit message. This
+   * function cannot be called until after a topic has been initialized
+   * in this service. Returns false if a save is not performed due to no
+   * changes pending, or true if otherwise. This function, upon success,
+   * will clear the UndoRedoService of pending changes. This function also
+   * shares behavior with setTopic(), when it succeeds.
+   */
+  saveTopic(
```

Comment:
This does an async operation and therefore we had used a `$rootScope.$applyAsync();`. but that has been removed. How are we handling this now?

Reply:
Made the function to return a promise. Every angularJS file where this function was called, I have added an $rootScope.$applyAsync();. This removes the requirement of using $rootScope.$applyAsync(); in the function itself